src/rep/mesh.jl

```diff
@@ -77,13 +77,19 @@ mutable struct TriMesh{T<:AbstractFloat,R<:Integer} <: AbstractMesh{T,R}
     _verts_len::AbstractArray{Int,1}
     _faces_len::AbstractArray{Int,1}
 
-    _verts_packed::Union{Nothing,AbstractArray{T,2}}
-    _verts_padded::Union{Nothing,AbstractArray{T,3}}
-    _verts_list::Union{Nothing,Vector{<:AbstractArray{T,2}}}
-
-    _faces_packed::Union{Nothing,Array{R,2}}
-    _faces_padded::Union{Nothing,Array{R,3}}
-    _faces_list::Union{Nothing,Vector{Array{R,2}}}
+    _verts_packed::AbstractArray{T,2}
```

Comment:
Don't store AbstractArray here. It can be CuArray/Array so best store it as `::R` and store R in the type signature. This will allow inference to work better on operations using these fields

Reply:
I have tried implementing storage type as type signature in struct, can you review it.

Reply:
The updated versions LGTM